src/ZfcRbac/Service/AuthorizationService.php

```diff
@@ -150,10 +177,6 @@ protected function assert($assertion, $context = null)
             return $assertion($this, $context);
         } elseif ($assertion instanceof AssertionInterface) {
             return $assertion->assert($this, $context);
-        } elseif (is_string($assertion)) {
```

Comment:
This is now back, you are right, it makes it more performant to have the retrieval only when it's called/needed.